pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -1107,13 +1110,9 @@ func (m *kubeGenericRuntimeManager) computeInitContainerActions(pod *v1.Pod, pod
 						if startup == proberesults.Failure {
 							// If the restartable init container failed the startup probe,
 							// restart it.
-							changes.ContainersToKill[status.ID] = containerToKillInfo{
-								name:      container.Name,
-								container: container,
-								message:   fmt.Sprintf("Init container %s failed startup probe", container.Name),
-								reason:    reasonStartupProbe,
-							}
-							changes.InitContainersToStart = append(changes.InitContainersToStart, i)
+							message = fmt.Sprintf("%s failed startup probe, will be restarted", message)
+							reason = reasonStartupProbe
+							restartContainer = true
 						}
 						break
```

Comment:
I still don't understand how we will get into restarting sidecar while it keep failing with the startup probe. This break will exit the loop so we never get to the `containerChanged` check. So I am not sure how test is working, left a comment there

Reply:
I don't think I understand the problem. When the container is restarted for any reason, it is restarted with the pod Spec. So whenever it is restarted, it will pick up the new image regardless of the initial reason for restarting.

Reply:
Does a container whose image has been updated restart after its startup probe got failed depending on its `periodSeconds` and `failureThreshold`? Wouldn't it be be better to restart the container soon after its image has been updated?